Src/PRuntimes/PExplicitRuntime/src/main/java/pexplicit/runtime/PExplicitGlobal.java

```diff
@@ -0,0 +1,116 @@
+package pexplicit.runtime;
+
+import lombok.Getter;
+import lombok.Setter;
+import pexplicit.commandline.PExplicitConfig;
+import pexplicit.runtime.machine.events.StateEvents;
+import pexplicit.runtime.scheduler.Scheduler;
+
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Represents global data structures represented with a singleton class
+ */
+public class PExplicitGlobal implements Serializable {
+    private static PExplicitGlobal PExplicitGlobal = null;
+    /**
+     * Singleton object of the class
+     **/
+    private static PExplicitConfig config = null;
+    /**
+     * Global configuration
+     **/
+    private static Scheduler scheduler = null;
+    /**
+     * Scheduler
+     **/
+    @Getter
+    @Setter
+    private static String status = "incomplete";
+    /**
+     * Status of the run
+     **/
+    @Getter
+    @Setter
+    private static String result = "error";
+    /**
+     * Result of the run
+     **/
+    private final Map<String, StateEvents> allStateEvents = new HashMap<>();    /* Mapping of each machine's state with its corresponding event handlers **/
```

Comment:
Why do we have this here?

Reply:
Good catch. This should be put at a different place, inside the State class object.
Will revise and confirm other fields in PExplicitGlobal class. Thanks!

Reply:
To be addressed in next PR.